clang/include/clang/Basic/DiagnosticParseKinds.td

```diff
@@ -1671,6 +1671,8 @@ def err_unexpected_module_decl : Error<
   "module declaration can only appear at the top level">;
 def err_module_expected_ident : Error<
   "expected a module name after '%select{module|import}0'">;
+def err_module_decl_cannot_be_macros : Error<
+  "module declaration cannot be a macro">;
```

Comment:
```suggestion
  "The name of a module|module partition declaration cannot contain a macro">;
```

Reply:
Should we also print the macro spelling text in diagnostic message? And I found that the first letter in other diagnostic message is lower case, should we update `The name of a ...` to `the name of a ...`?